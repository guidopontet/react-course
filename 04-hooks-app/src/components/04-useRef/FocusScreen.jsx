import React, { useRef } from 'react'

export const FocusScreen = () => {

  const inputRef = useRef();

  const handleClick = () => {
    inputRef.current.select();
  }

  return (
    <div className='container my-3'>
      <h1>FocusScreen</h1>
      <hr />

      <input ref={ inputRef }
             type="text"
             className='form-control'
             placeholder='Nombre' />

      <button className='btn btn-outline-primary my-3'
              onClick={ handleClick }>Focus</button>
    </div>
  )
}
