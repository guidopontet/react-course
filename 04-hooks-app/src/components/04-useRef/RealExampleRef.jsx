import React, { useState } from 'react'

import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";

export const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  return (
    <div className='container my-3'>
      <h1>RealExampleRef</h1>
      <hr />

      { show && <MultipleCustomHooks />}

      <button className='btn btn-primary my-5'
              onClick={ () => setShow(!show) }>Mostrar / Ocultar</button>
    </div>
  )
}
