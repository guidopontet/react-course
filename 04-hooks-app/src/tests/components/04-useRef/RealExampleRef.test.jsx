import { shallow } from 'enzyme';

import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('RealExampleRef', () => {
  const wrapper = shallow(<RealExampleRef />);

  test('should render', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('should render MultipleCustomHooks component', () => {
    wrapper.find('button').simulate('click');

    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  })
})
