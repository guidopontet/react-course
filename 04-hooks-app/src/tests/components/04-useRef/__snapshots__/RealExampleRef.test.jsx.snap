// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExampleRef should render 1`] = `
<div
  className="container my-3"
>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary my-5"
    onClick={[Function]}
  >
    Mostrar / Ocultar
  </button>
</div>
`;
